CS3230Project/View/SearchPatient.cs

```diff
@@ -0,0 +1,115 @@
+﻿using CS3230Project.Model.Accounts;
+using System;
+using System.Collections.Generic;
+using System.Windows.Forms;
+using CS3230Project.Model.Users.Patients;
+
+namespace CS3230Project.View
+{
+    /// <summary>
+    ///   The search patient view
+    /// </summary>
+    public partial class SearchPatient : Form
+    {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="SearchPatient" /> class.
+        /// </summary>
+        public SearchPatient()
+        {
+            this.InitializeComponent();
+            this.bindLabelsToCurrentUser();
+        }
+
+        private void bindLabelsToCurrentUser()
+        {
+            this.loggedInAsLabel.Text = $"Logged In As: {CurrentUser.User.UserName}";
+            this.userIdLabel.Text = $"User ID: {CurrentUser.User.Id}";
+            this.nameLabel.Text = $"Name: {CurrentUser.User.FirstName} {CurrentUser.User.LastName}";
+        }
+
+        private void logoutButton_Click(object sender, EventArgs e)
+        {
+            CurrentUser.User = null;
+            Form homeForm = new Login();
+            homeForm.Location = Location;
+            homeForm.StartPosition = FormStartPosition.Manual;
+            homeForm.FormClosing += delegate { Show(); };
+            Hide();
+            homeForm.Size = this.Size;
+            homeForm.ShowDialog();
+            Close();
+        }
+
+        private void searchButton_Click(object sender, EventArgs e)
+        {
+            List<Patient> matchingPatients = new List<Patient>();
+            if (this.searchFirstAndLastNameCheckBox.Checked && this.searchByBirthDateCheckBox.Checked)
```

Comment:
Is holding up approval. Sorry I did not catch this on my first look, but you do need try catch blocks around this code. When the user clicks the search patient button with empty fields the application crashes because it does not have try catch blocks. So basically the whole method should be wrapped in try catch blocks like this.

            try
            {
                List<Patient> matchingPatients = new List<Patient>();
                if (this.searchFirstAndLastNameCheckBox.Checked && this.searchByBirthDateCheckBox.Checked)
                {
                    matchingPatients = PatientManager.GetPatientsByNameAndDateOfBirth(this.firstNameTextBox.Text,
                        this.lastNameTextBox.Text, this.dateOfBirthDatePicker.Value);
                }
                else if (this.searchFirstAndLastNameCheckBox.Checked)
                {
                    matchingPatients =
                        PatientManager.GetPatientsByName(this.firstNameTextBox.Text, this.lastNameTextBox.Text);
                }
                else if (this.searchByBirthDateCheckBox.Checked)
                {
                    matchingPatients = PatientManager.GetPatientsByDateOfBirth(this.dateOfBirthDatePicker.Value);
                }

                this.displayPatientData(matchingPatients);
            }
            catch (ArgumentException exception)
            {
                MessageBox.Show(exception.Message, "SAMPLE ERROR");
            }

Reply:
Fixed. Good catch!